piff/gp_interp.py

```diff
@@ -0,0 +1,324 @@
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: sklearn_gp_interp
+"""
+
+from .interp import Interp
+from .star import Star, StarFit
+
+import numpy as np
+
+class GPInterp(Interp):
+    """
+    An interpolator that uses sklearn.gaussian_process to interpolate a single surface.
+
+    :param  attr_interp: A list of star attributes to interpolate from
+    :param  kernel:      A Kernel object indicating how the different stars are correlated.
+    :param  optimizer:   Optimizer to use for optimizing the kernel.  Set to `None` to skip
+                         kernel optimization.
+    :param  npca:        Number of principal components to keep.  Set to `0` to skip decomposition
+                         of PSF model into principal components.
+    :param  logger:      A logger object for logging debug info. [default: None]
+    """
+    def __init__(self, attr_interp=None, kernel=None, optimizer='fmin_l_bfgs_b', npca=0,
+                 logger=None):
+        from sklearn.gaussian_process import GaussianProcessRegressor
+        if isinstance(kernel, str):
+            kernel = self._eval_kernel(kernel)
+
+        if attr_interp is None:
+            attr_interp = ['u', 'v']
+
+        self.attr_interp = attr_interp
+        self.kernel = kernel
+        self.npca = npca
+
+        self.kwargs = {
+            'attr_interp': attr_interp,
+            'optimizer': optimizer,
+            'npca': npca,
+            'kernel': repr(kernel)
+        }
+
+        self.gp = GaussianProcessRegressor(self.kernel, optimizer=optimizer)
+
+    @staticmethod
+    def _eval_kernel(kernel):
+        from sklearn.gaussian_process.kernels import WhiteKernel, RBF, Matern, RationalQuadratic
+        from sklearn.gaussian_process.kernels import ExpSineSquared, DotProduct, PairwiseKernel
+        from numpy import array
+        return eval(kernel)
```

Comment:
Should there be some sort of error catching to say what valid kernels are available? I can imagine it would be really frustrating if something types 'ExpSinSquared' and doesn't understand why it's failing...